tests/run.sh

```diff
@@ -13,10 +13,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -eu
+if [ ! -z "$TEST_NAME" ]; then
+  SELECTED_TEST_NAME=$TEST_NAME
+else
+  SELECTED_TEST_NAME=$(find tests -name run.sh | grep -v tests/run.sh | sort | cut -d/  -f2 | tr '\r\n' ' ')
```

Comment:
```suggestion
  SELECTED_TEST_NAME=$(find tests -mindepth 2 -maxdepth 2 -name run.sh | cut -d/ -f2)
```

no need to convert new lines into spaces.

btw the whole thing can be simply

```sh
SELECTED_TEST_NAME="${TEST_NAME-$(find tests -mindepth 2 -maxdepth 2 -name run.sh | cut -d/ -f2)}"
```